nixos/modules/hardware/video/nvidia.nix

```
@@ -176,7 +176,9 @@ in
 
     hardware.nvidia.package = lib.mkOption {
       type = types.package;
-      default = config.boot.kernelPackages.nvidiaPackages.stable;
+      default = with config.boot.kernelPackages.nvidiaPackages; if config.nixpkgs.system != "i686-linux"
+        then stable
+        else legacy_390;
```

```suggestion
      default = config.boot.kernelPackages.nvidiaPackages.stable;
```
Instead of checking here for "i686-linux" would be better to check where currently has the warning, so the warning can be removed, and proper package suggested instead.